src/import/FSHImporter.ts

```diff
@@ -1009,7 +1040,7 @@ export class FSHImporter extends FSHVisitor {
   visitCardRule(ctx: pc.CardRuleContext): (CardRule | FlagRule)[] {
     const rules: (CardRule | FlagRule)[] = [];
 
-    const cardRule = new CardRule(this.visitPath(ctx.path()))
+    const cardRule = new CardRule(this.getPathWithContext(this.visitPath(ctx.path()), ctx))
```

Comment:
Yeah I made this choice because I was worried about what would happen if I did bury this in `visitPath`, and then we started using it to get context on things that aren't paths, namely CodeSystems. I thought just operating on the string would give us the most flexibility. But then again the current `prependPathContext` function definitely assumes we are generating a `.` separated path, so things will have to change anyway.

Reply:
I'm now remembering another issue I had with trying to do everything in `visitPath`. The problem is that sometimes the `PathContext` is undefined, so in that case I wasn't able to get the parent off of it, and had to pass in the `parentCtx` directly, which is not allowed because now I am messing with `visitPath`'s signature. I'm not sure how to get around that. I could change `visitPath` to accept the parentCtx's directly, and pull the `path()` off of that, but that feels a little suspect.

Reply:
Yeah, the possibility of an undefined context throws a wrench in this idea.  Sorry for the distraction!